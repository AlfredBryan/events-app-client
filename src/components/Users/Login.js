import React, { Component } from "react";
import { Link } from "react-router-dom";
import axios from "axios";

const apiURL = "https://events-apps.herokuapp.com/api/user/login";

class Login extends Component {
  constructor() {
    super();
    this.state = {
      email: "",
      password: "",
      errorMessage: ""
    };
  }

  handleSubmit = e => {
    e.preventDefault();
    let { email, password } = this.state;
    axios
      .post(`${apiURL}`, { email, password })
      .then(res => {
        if (res.status === 200) {
          localStorage.setItem("token", res.data.token);
          this.props.history.replace("/home");
        }
      })
      .catch(error => {
        if (error) {
          this.setState({
            errorMessage: "username or password incorrect"
          });
        }
      });
  };

  handleChange = e => {
    this.setState({
      [e.target.name]: e.target.value
    });
  };

  render() {
    return (
      <React.Fragment>
        <div className="container register">
          <div className="row">
            <div className="col-md-3 register-left">
              <img src="https://image.ibb.co/n7oTvU/logo_white.png" alt="" />
              <h3>Welcome</h3>
              <Link to="/" style={{ color: "white", textDecoration: "none" }}>
                Register
              </Link>
            </div>
            <div className="col-md-9 register-right">
              <h3 className="register-heading">Login</h3>
              <div className="row register-form">
                <div className="col-md-12 profile_card">
                  <form
                    onSubmit={this.handleSubmit}
                    encType="multipart/form-data"
                    className="form-inline"
                  >
                    <div className="form-group">
                      <i className="fa fa-envelope-o" />
                      <input
                        type="text"
                        name="email"
                        className="form-control"
                        placeholder="Email"
                        id="email"
                        value={this.state.email}
                        onChange={this.handleChange}
                      />
                    </div>
                    <div className="form-group">
                      <i className="fa fa-lock" />
                      <input
                        type="password"
                        name="password"
                        id="password"
                        className="form-control"
                        placeholder="Password *"
                        value={this.state.password}
                        onChange={this.handleChange}
                      />
                    </div>
                    <div className="float-right">
                      <button
                        onClick={this.handleSubmit}
                        type="submit"
                        className="btn btn-primary"
                        value="Register"
                      >
                        Login
                      </button>
                    </div>
                  </form>
                </div>
              </div>
            </div>
          </div>
          <p style={{ color: "white", fontSize: "20px" }}>
            {this.state.errorMessage}
          </p>
        </div>
      </React.Fragment>
    );
  }
}

export default Login;
