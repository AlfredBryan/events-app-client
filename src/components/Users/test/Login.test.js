import React from "react";
import { shallow } from "enzyme";
import Login from "../Login";

describe("Login Component", () => {
  it("should render without throwing an error", () => {
    expect(
      shallow(<Login />)
        .find("form.login")
        .exists()
    ).toMatchSnapshot();
  });
  it("renders a email input", () => {
    expect(shallow(<Login />).find("#email").length).toMatchSnapshot();
  });
  it("renders a password input", () => {
    expect(shallow(<Login />).find("#password").length).toMatchSnapshot();
  });
});

describe("Email input", () => {
  it("should respond to change event and change the state of the Login Component", () => {
    const wrapper = shallow(<Login />);
    wrapper.find("#email").simulate("change", {
      target: { name: "email", value: "test@gmail.com" }
    });

    expect(wrapper.state("email")).toMatchSnapshot("test@gmail.com");
  });
});

describe("Password input", () => {
  it("should respond to change event and change the state of the Login Component", () => {
    const wrapper = shallow(<Login />);
    wrapper.find("#password").simulate("change", {
      target: { name: "password", value: "testp" }
    });

    expect(wrapper.state("password")).toEqual("testp");
  });
});
