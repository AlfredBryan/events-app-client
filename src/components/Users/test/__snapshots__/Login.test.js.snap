// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Email input should respond to change event and change the state of the Login Component: test@gmail.com 1`] = `"test@gmail.com"`;

exports[`Login Component renders a email input 1`] = `1`;

exports[`Login Component renders a password input 1`] = `1`;

exports[`Login Component should render without throwing an error 1`] = `false`;
